### PR TITLE
increase clang-tidy coverage

### PR DESCRIPTION
**Pull request by Piotrek321**

### CODE REVIEW

File: src/api/session/OTX.cpp [hunk 1]
```diff
@@ -169,8 +170,8 @@ namespace opentxs::api::session::imp
 OTX::OTX(
     const Flag& running,
     const api::session::Client& client,
-    const ContextLockCallback& lockCallback)
-    : lock_callback_(lockCallback)
+    ContextLockCallback lockCallback)
```
Comment (wiesiekpap) [inline src/api/session/OTX.cpp:173]: Why not pass it as rvalue? At present, a copy is forced on calling this.

File: src/blockchain/database/common/Database.cpp [hunk 2]
```diff
@@ -219,7 +219,7 @@ struct Database::Imp {
             throw std::runtime_error("Failed to construct path");
         }
 
-        std::ofstream{v1.string()};
+        std::ofstream give_me_a_name{v1.string()};
```
Comment (wiesiekpap) [inline src/blockchain/database/common/Database.cpp]: Might as well use the name to check it for open, as file creation is the only thing this really achieves.

File: src/blockchain/p2p/peer/Peer.cpp [hunk 3]
```diff
@@ -791,7 +792,7 @@ auto Peer::transmit(zmq::Message&& message) noexcept -> void
     }
 
     if (result) {
-        log_(OT_PRETTY_CLASS())("Sent ")(payload.size())(" bytes").Flush();
+        log_(OT_PRETTY_CLASS())("Sent ")(payloadBytes)(" bytes").Flush();
```
Comment (wiesiekpap) [inline src/blockchain/p2p/peer/Peer.cpp:795]: good catch

**Review by wiesiekpap — APPROVED** [review]

File: src/api/session/OTX.cpp (hunk 1 above)
Comment (Piotrek321) [reply src/api/session/OTX.cpp:173]: this is std::function. As far as I know there is no difference in passing by value by reference (I might be wrong though)

Comment (Piotrek321) [reply src/api/session/OTX.cpp:173]: also in this case contst & was removed (at least I think so) becasue this callback is moved one line under.

**Pull request merged**